### PR TITLE
On demand to ri ec2 report

### PR DESCRIPTION
**Pull request by zanchi-r**
This PR adds the on demand to RI ec2 report generation.
A new task fetch-pricings was created to retrieve the pricings and should be run by nomad every 24 hours.
The report is generated during the process-account task and saved into ES.

### CODE REVIEW

File: aws/pricings/fetchEc2Pricings.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+)
+
+var (
+	EC2ServiceCode = "AmazonEC2"
+	// Pricing endpoints are only available in "us-east-1" and "ap-south-1"
+	PricingApiEndpointRegion = "us-east-1"
+)
+
+// Specs stores the cost specifications for an instance type
+type Specs struct {
```
Comment (thibautcornolti) [inline aws/pricings/fetchEc2Pricings.go]: The struct name should be more precise.

File: aws/pricings/fetchEc2Pricings.go [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+	ThreeYearsStandardNoUpfrontHourlyCost float64 `json:"threeYearsStandardNoUpfrontHourlyCost"`
+}
+
+// Type maps an instance type to a Specs struct
+type Type struct {
+	Type map[string]*Specs `json:"type"`
+}
+
+// Platform maps a platform to a Type struct
+type Platform struct {
```
Comment (thibautcornolti) [inline aws/pricings/fetchEc2Pricings.go]: Again

File: aws/pricings/fetchEc2Pricings.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+// Specs stores the cost specifications for an instance type
+type Specs struct {
+	CurrentGeneration                     bool    `json:"currentGeneration"`
+	OnDemandHourlyCost                    float64 `json:"onDemandHourlyCost"`
+	OneYearStandardNoUpfrontHourlyCost    float64 `json:"oneYearStandardNoUpfrontHourlyCost"`
+	ThreeYearsStandardNoUpfrontHourlyCost float64 `json:"threeYearsStandardNoUpfrontHourlyCost"`
+}
+
+// Type maps an instance type to a Specs struct
+type Type struct {
```
Comment (thibautcornolti) [inline aws/pricings/fetchEc2Pricings.go]: Again

File: aws/pricings/fetchEc2Pricings.go [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+	} else if instanceType, ok := attributes["instanceType"]; ok == false {
+	} else {
+		return instanceType.(string)
+	}
+	return ""
+}
+
+// isCurrentGeneration takes an item from the aws json pricing and returns
+// true if it is a current generation instance
+func isCurrentGeneration(item aws.JSONValue) *bool {
```
Comment (thibautcornolti) [inline aws/pricings/fetchEc2Pricings.go]: Why did you use a pointer here?

**Review by thibautcornolti — CHANGES_REQUESTED** [review]

**Review by thibautcornolti — APPROVED** [review]

**Review by thibautcornolti — APPROVED** [review]

**Review by hug33k — APPROVED** [review]

**Pull request merged**